### PR TITLE
Fix issue #512

### PR DESCRIPTION
**Pull request by nrybowski**
This PR proposes a fix for issue #512.

### CODE REVIEW

**Comment by anthonygego** [comment]
It would be worth adding those lines for external authentication systems as well.

**Comment by nrybowski** [comment]
Rebase on 60f10cb4c375ce207471043e76bd813220b95399 and add required updates.

**Comment by nrybowski** [comment]
Restore updates due to pymongo bump.

**Review by anthonygego — CHANGES_REQUESTED** [review]
I'd rather put this in a small utility function to ease the maintenance.

**Comment by nrybowski** [comment]
Rebase on b5794b89dba6f3f428ece255e7f354cde23c8ef8 and move email sanitation routine in `user_manager`.

File: inginious/frontend/user_manager.py [hunk 1]
```diff
@@ -19,6 +19,23 @@
 import pymongo
 from binascii import hexlify
 import os
+import re
+
+
+def sanitize_email(email: str) -> str:
```
Comment (Drumor) [inline inginious/frontend/user_manager.py:25]: Isn't it better to set this as a classmethod within the UserManager class?

Comment (nrybowski) [reply inginious/frontend/user_manager.py]: Fixed in a9bbbaa903b5fbb0e1533ac13c416393cf4a5a78.

**Comment by nrybowski** [comment]
Address review and rebase on 134371f45cc9cf4a4805c0ec85501c1308691919.

**Review by Drumor — COMMENTED** [review]
Looks good to me.

File: inginious/frontend/installer.py [hunk 2]
```diff
@@ -550,7 +551,14 @@ def configure_authentication(self, database):
 
         username = self._ask_with_default("Enter the login of the superadmin", "superadmin")
         realname = self._ask_with_default("Enter the name of the superadmin", "INGInious SuperAdmin")
-        email = self._ask_with_default("Enter the email address of the superadmin", "superadmin@inginious.org")
+        while True:
```
Comment (anthonygego) [inline inginious/frontend/installer.py:554]: email = None
      while not email:
         ...  
will require one line less and be more pleasant to read.

**Comment by nrybowski** [comment]
Rebase on df39f703d98922476be8a9338203a5eb3b1d2ac2 and address review.

File: inginious/frontend/installer.py (hunk 2 above)
Comment (nrybowski) [reply inginious/frontend/installer.py]: Fixed in 3da81fd3f7509961d623351c5002dc8a9f583053.

**Comment by nrybowski** [comment]
I think there is no interest in keeping separate commits for this PR, they could all be squashed in one.

**Comment by anthonygego** [comment]
LGTM

**Pull request merged**